src/epanet.c

```diff
@@ -2830,22 +2888,19 @@ int DLLEXPORT EN_getbasedemand(EN_Project p, int nodeIndex, int demandIndex,
 **----------------------------------------------------------------
 */
 {
+    Pdemand d;
+
     // Check for valid arguments
+    *baseDemand = 0.0;
     if (!p->Openflag) return 102;
     if (nodeIndex <= 0 || nodeIndex > p->network.Nnodes) return 203;
 
-    // Retrieve demand for specified category
-    if (nodeIndex <= p->network.Njuncs)
-    {
-        // Locate demand category record and assign demandName to it
-        list_t *dlist = p->network.Node[nodeIndex].D;
-        list_node_t *lnode = get_nth_list(dlist, demandIndex);
-        if (!lnode)
-            return 253;
-        else
-            *baseDemand = get_base_demand(lnode) * p->Ucf[FLOW];
-    }
-    else *baseDemand = (double)(0.0);
+    // Locate target demand in node's demands list
+    d = finddemand(p->network.Node[nodeIndex].D, demandIndex);
+    if (d == NULL) return 253;
```

Comment:
not sure if we ever settled on a code style guide... but these single-line conditional/returns are so frustrating when I want to place a breakpoint. I'm completely fine moving forward with this PR as-is, but let's make a determination about style soon 👍